File: dungeonGame.py
#File Name: dungeonGameApp.py
#This is the application which runs dungeonGame
#Author: Bryant P.

import random

#Setup
class Monster():
    def __init__(self, health, minDmg, maxDmg, maxChance):    
        self.health = health
        self.minDmg = minDmg
        self.maxDmg = maxDmg
        self.maxChance = maxChance
    
    def takeDmg(self, dmg):
        self.health -= dmg
        print('You deal', dmg, 'damage.\nThe monster now has',self.health, 'health.')

    def chanceEvent(self):
        self.chance = random.randint(0, self.maxChance)
        return self.chance

class Player():
    def __init__(self, health, minDmg, maxDmg, inventory, run, choice):
        self.health = health
        self.minDmg = minDmg
        self.maxDmg = maxDmg
        self.inventory = inventory
        self.run = run
        self.choice = choice

    def takeDmg(self, dmg):
        self.health -= dmg
        print('The monster attacks, dealing', dmg, 'damage.\nYou now have',self.health, 'health.')
    
    def useItem(self, item):
        if item in range(len(self.inventory)+1): #Make inventory with dictionary
            self.health += 40
            print('You heal for 40 HP.')
            self.inventory.remove(self.inventory[item-1])
        else:
            print('You have none of that item.')
 
def choiceEvent(player, monster):
    player.choice = int(input('Pick a number:\n(1) Attack\n(2) Inventory\n(3) Run\n'))

def choiceLogic(player, monster):
        if player.choice == 1:
            atkEvent(player, monster)
        elif player.choice == 2:
            if len(player.inventory) > 0:
                print('(1)',player.inventory[0])
                item = int(input('Choose one.\n'))
                player.useItem(item)
            else:
                print('You have no items left.')
        elif player.choice == 3:
            player.run = True
        if player.run == False:
            if monster.chanceEvent() == 0: #Chance not working
                print('The monster is too tired to move.')
            else:
                atkEvent(monster, player)

def atkEvent(attacker, defender):
    dmg = random.randint(attacker.minDmg, attacker.maxDmg)
    defender.takeDmg(dmg)

def endingEvent(player, monster):
    if player.health >= 0 and monster.health < 0:
        print('You have slain the monster!\nGo collect the treasure!')
    elif player.health <= 0 and monster.health > 0:
        print('You have been bested by the monster.\nYou get no treasure.')
    elif player.run == True:
        print('You escape the dungeon with your life,\nbut not the treasure.')
    else:
        print('Humorously, you defeat eachother.\nNo one gets the treasure.')
#Prompts
def main():
    player1 = Player(125, 20, 30, ['Health Potion'], False, 0)
    monster1 = Monster(150, 25, 35, 2)

    print('You enter the dungeon.')
    print('You encounter a monster.\nIt is gaurding a hoarde of treasure.')

    while player1.health > 0 and monster1.health > 0 and player1.run == False:
        choiceEvent(player1, monster1)
        choiceLogic(player1, monster1)
    endingEvent(player1, monster1)
#Game
main()
    
    

File: 06_Programming/dungeonGame/dungeonGame.js
// File Name: dungeonGame.js
//
// This is a text-based adventure based in a dungeon.
//
// Author: Bryant Phan

//Variables
const readline = require('readline-sync')
const chalk = require('chalk');

let monstHP = 125
let playerHP = 100
let atkDmg = 0
let healthPotion = 30
let chance = 0
let item = ''

//Functions
function playerAtk(atkDmg, monstHP) {
    atkDmg = Math.ceil(Math.random() * 11) + 10
    monstHP -= atkDmg
    console.log(chalk.blueBright('You ') +  'swing Your sword.')
    console.log(chalk.blueBright('You ') + 'deal ' + chalk.red(atkDmg) + ' damage to ' + chalk.green('The Monster') + '.')
    if (monstHP > 0) {
        console.log(chalk.green('The Monster ') +  'now has ' + chalk.red(monstHP) + ' health.')
    }
    return monstHP
}
function inventory(item, playerHP, healthPotion) {
    item = readline.question('Choose One:\nHealth Potion' + '\n')
    if (item == 'Health Potion') {
        playerHP += healthPotion
        console.log(chalk.blueBright('You ') + 'now have ' + chalk.red(playerHP) + ' health.')
    }
    else {
        console.log('That is not an item.')
    }
    return playerHP
}
function monstAtk(monstHP, chance, atkDmg, playerHP) {
    if (monstHP > 0) {
        chance = Math.ceil(Math.random() * 5)
        if (chance != 4) {
            console.log(chalk.green('The Monster ') +  'attacks.')
            atkDmg = Math.ceil(Math.random() * 11) + 15
            playerHP -= atkDmg
            console.log(chalk.blueBright('You ') + 'take ' + chalk.red(atkDmg) + ' Damage')
            if (playerHP > 0) {
                console.log(chalk.blueBright('You ') + 'now have ' + chalk.red(playerHP) + ' health.')
            }
            else {
                console.log(chalk.blueBright('You ') + 'have no health left')
            }
        }
        else{
            console.log(chalk.green('The Monster ') +  'is too tired to move.')
        }
    }
    return playerHP
}

module.exports = {
    playerAtk: playerAtk,
    inventory: inventory,
    monstAtk: monstAtk,
    monstHP: monstHP,
    playerHP: playerHP,
    atkDmg: atkDmg,
    healthPotion: healthPotion,
    chance: chance,
    item: item
}